tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs

```diff
@@ -98,22 +101,22 @@ public AzPredictorService(string serviceUri, ITelemetryClient telemetryClient, I
             Validation.CheckArgument(telemetryClient, $"{nameof(telemetryClient)} cannot be null.");
             Validation.CheckArgument(azContext, $"{nameof(azContext)} cannot be null.");
 
-            _commandsEndpoint = $"{serviceUri}{AzPredictorConstants.CommandsEndpoint}?clientType={AzPredictorService.ClientType}&context={JsonSerializer.Serialize(new CommandRequestContext(), JsonUtilities.DefaultSerializerOptions)}";
+            _commandsEndpoint = $"{serviceUri}{AzPredictorConstants.CommandsEndpoint}?clientType={AzPredictorService.ClientType}&context.versionNumber={azContext.AzVersion}";
             _predictionsEndpoint = serviceUri + AzPredictorConstants.PredictionsEndpoint;
             _telemetryClient = telemetryClient;
             _azContext = azContext;
 
             _client = new HttpClient();
 
-            RequestAllPredictiveCommands();
+            RequestAllPredictiveCommands(this._azContext.AzVersion);
         }
 
         /// <summary>
         /// A default constructor for the derived class. This is used in test cases.
         /// </summary>
         protected AzPredictorService()
         {
-            RequestAllPredictiveCommands();
+            RequestAllPredictiveCommands(this._azContext?.AzVersion);
```

Comment:
Doesn't need `?`. The constructor checks to make sure that it's not `null` (line 102). Or just pass in `null` here since `_azContext` never be able to initialized from this code path.